internal/storage/postgres.go

```diff
@@ -0,0 +1,310 @@
+    password VARCHAR(50)
+);
+CREATE TABLE IF NOT EXISTS orders (
+				  orderID           VARCHAR(64) UNIQUE,
+				  userID           INT NOT NULL,
+				  state 	  TEXT NOT NULL,
+				  accrual	double precision ,
+					withdraw double precision,
+					uploaded_at TIMESTAMP );
+
+CREATE TABLE IF NOT EXISTS balance (
+				userID           INT UNIQUE NOT NULL,
+				balance double precision);
+
+CREATE TABLE IF NOT EXISTS withdrawals  (
+				userID				INT NOT NULL,
+  				orderID				VARCHAR(64) UNIQUE,
+				sum					double precision,
+    			processed_at		TIMESTAMP);
+
+`
+	_, err = conn.Exec(context.Background(), query)
+	if err != nil {
+		log.Println(err)
+		return &PostgresStorage{Connection: conn}, conn, err
+	}
+
+	return &PostgresStorage{Connection: conn}, conn, nil
+}
+
+func (p *PostgresStorage) RegisterUser(credentials Credentials) error {
+
+	query := `
+		INSERT INTO users(
+		username,
+		password
+		)
+		VALUES($1, $2);
+	`
+	_, err := p.Connection.Exec(context.Background(), query, credentials.Username, credentials.Password)
+	if err != nil {
+		log.Println(err)
+		return err
+
+	}
+	return nil
+}
+
+func (p *PostgresStorage) SelectUser(user Credentials) (Credentials, error) {
+
+	query := `
+		SELECT ID FROM users
+		WHERE username like $1;
+	`
+	_, err := p.Connection.Exec(context.Background(), query, user.Username)
+	if err != nil {
+		log.Println(err)
+		return user, err
+
+	}
+	return user, nil
+}
+func (p *PostgresStorage) SignIn(user Credentials) (Credentials, error) {
+	var id int
+	query := `
+		SELECT id FROM users WHERE username=$1 and password=$2;
+	`
+	result, err := p.Connection.Query(context.Background(), query, user.Username, user.Password)
+
+	if err != nil {
+		return user, err
+	}
+	defer result.Close()
+	for result.Next() {
+		err = result.Scan(&id)
+
+	}
+	user.ID = id
+	if err != nil {
+		return user, errors.New("not auth")
+	} else {
+		return user, nil
+	}
+}
+func (p *PostgresStorage) PostOrder(order Order) error {
+
+	query := `INSERT INTO orders(
+					orderID, userID, state, accrual, uploaded_at
+					)
+					VALUES($1, $2, $3, 0, $5)
+ON CONFLICT (orderID) DO 
+		    UPDATE SET 	state=$3,
+		            	accrual=$4
+		            	;`
+	res, err := p.Connection.Exec(context.Background(), query, order.OrderID, order.UserID, order.State, order.Accrual, order.UploadedAt)
+	if err != nil {
+		log.Println(err)
+		return err
+
+	}
+	log.Println(res)
+	return nil
+}
+
+func (p *PostgresStorage) GetOrderByOrderID(id string) (Order, error) {
+	var order Order
+	order.OrderID = id
+	query := `
+		SELECT orderID, userID, state, accrual, uploaded_at FROM orders  WHERE orderID=$1;
+	`
+	result, err := p.Connection.Query(context.Background(), query, id)
+
+	if err != nil {
+		return order, err
+	}
+	defer result.Close()
+	for result.Next() {
+		err = result.Scan(&order.OrderID, &order.UserID, &order.State, &order.Accrual, &order.UploadedAt)
+
+	}
+
+	if err != nil {
+		return order, err
+	} else {
+		return order, nil
+	}
```

Comment:
лишнее ветвление, можно просто
```suggestion
	} 
	return order, nil
	
```